### PR TITLE
#165273485 Add article tags feature

### PR DESCRIPTION
**Pull request by sanya-kenneth**
### What does this PR do?
- Add create article taglist functionality
- Add get article taglist functionality
- Add update article taglist functionality
- Add get tags functionality

###### Create tags 
`[POST] - /api/articles/`

`Request body`
```
{
	"title": "The new era of Tech",
	"description": "Cool tech to try out",
	"body": "Watch as tech helps change many lives around the globe",
	"tagList": ["tech", "technology", "techjournal", "techbeats"]
}
```
`Response body`
```
{
    "article": {
        "article": {
            "id": 48,
            "title": "The new era of Tech",
            "description": "Cool tech to try out",
            "body": "Watch as tech helps change many lives around the globe",
            "tagList": ["tech", "technology", "techjournal", "techbeats"],
            "slug": "the-new-era-of-tech-nume",
            "created_at": "2019-05-02T10:07:09.230504Z",
            "updated_at": "2019-05-02T10:07:09.230515Z",
            "favorited": false,
            "favorite_count": 0,
            "author": "admin"
        }
    }
}
```

##### Edit Tag List
`PUT - /api/articles/<int:article_id>/`

`Request body`

```
{
	"title": "The new era of Tech",
	"description": "Cool tech to try out",
	"body": "Watch as tech helps change many lives around the globe",
	"tagList": ["techcrunch blog"]
}
```
`Response body`
```
{
    "article": {
        "article": {
            "id": 48,
            "title": "The new era of Tech",
            "description": "Cool tech to try out",
            "body": "Watch as tech helps change many lives around the globe",
            "tagList": ["techcrunch blog"],
            "slug": "the-new-era-of-tech-nume",
            "created_at": "2019-05-02T10:07:09.230504Z",
            "updated_at": "2019-05-02T10:07:09.230515Z",
            "favorited": false,
            "favorite_count": 0,
            "author": "admin"
        }
    }
}
```

##### Get Tags
`GET -  /api/tags/`

```
{
  "tags": [
          "tech",
           "tech-space"
       ]
}
```
### Relevant stories
[#165273485](https://www.pivotaltracker.com/story/show/165273485)

### CODE REVIEW

**Comment by archibishop** [comment]
@sanya-kenneth I am unable to update the tag list for an article. Can you look into this.

File: authors/apps/article_tags/serializers.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from rest_framework import serializers
+from .models import ArticleTag
+
+
+class ArticleTagSerializer(serializers.ModelSerializer):
```
Comment (3Nakajugo) [inline authors/apps/article_tags/serializers.py:5]: please add a doc string to describe what the class does

File: authors/apps/article_tags/tests/base.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from rest_framework.test import APIClient, APITestCase
+from django.urls import reverse
+# Local import
+from .test_data import TestData
+from authors.apps.authentication.models import User
+
+
+class BaseTestCase(APITestCase):
```
Comment (3Nakajugo) [inline authors/apps/article_tags/tests/base.py:8]: some thing here. please add a description for the class

File: authors/apps/article_tags/serializers.py (hunk 1 above)
Comment (sanya-kenneth) [reply authors/apps/article_tags/serializers.py:5]: I have added a doc string.. Thanks for the observation

File: authors/apps/article_tags/tests/base.py (hunk 2 above)
Comment (sanya-kenneth) [reply authors/apps/article_tags/tests/base.py:8]: I have added a doc string.. Thanks for the observation

File: authors/apps/article_tags/tests/test_article_tag.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        create_url = reverse('articles-list-create')
+        update_url = '/api/articles/3/'
+        res = self.client.post(
+            create_url, self.data.create_arcticle, HTTP_AUTHORIZATION=self.auth_header,
+            format="json")
+        response = self.client.put(
+            update_url, self.data.empty_tag_list, HTTP_AUTHORIZATION=self.auth_header,
+            format="json") 
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.data['error'], 'Tag field can not be an empty list')
```
Comment (engjames) [inline authors/apps/article_tags/tests/test_article_tag.py]: Add a new line after this statement

Comment (sanya-kenneth) [reply authors/apps/article_tags/tests/test_article_tag.py]: Thanks for the observation. I have added a new line to at the end of the file.

**Review by 3Nakajugo — APPROVED** [review]

**Review by mwinel — APPROVED** [review]
@archibishop as per the implementation requirements, this is ready for a merge.

**Review by archibishop — APPROVED** [review]
@sanya-kenneth LGTM I am merging this.

**Pull request merged**